CryptoPrices/Sources/Styleguide/Sources/Styleguide/FormatStyle/DollarCurrencyFormatStyle.swift

```diff
@@ -0,0 +1,17 @@
+//
+//  DollarCurrencyFormatStyle.swift
+//  Styleguide
+//
+//  Created by Doan Thieu on 15/12/2022.
+//
+
+import Foundation
+
+// We want the "$" sign always prefix the amount (e.g "$ 1,200"),
+// so we can't use built-in Decimal.FormatStyle.Currency
+public struct DollarCurrencyFormatStyle: FormatStyle {
+
+    public func format(_ value: Decimal) -> String {
+        return "$ \(value.formatted(.number.precision(.fractionLength(0...2))))"
```

Comment:
Should we use a [NumberFormatter](https://developer.apple.com/forums/thread/693152) here to support displaying currency value naturally and with localization instead? 
<img width="1094" alt="Screenshot 2022-12-19 at 15 21 34" src="https://user-images.githubusercontent.com/70877098/208380048-64b37d57-9c60-48cb-8275-22fee386f306.png">


P/S: I am not sure why do we need a space between the dollar sign and the value in the current format to be honest 🤔

Reply:
- About the space, it's accidentally added, will remove it 🙏 
- As indicated in the header:
   ```
   // We want the "$" sign always prefix the amount (e.g "$ 1,200"),
   // so we can't use built-in Decimal.FormatStyle.Currency
   ```
the design always prefixes "$" regardless of locale, so we can't use the built-in one for currency. We have to format the amount as a number (which follows locale correctly with the decimal characters`.` and `,`) and then prefix it with the dollar sign `$`
Using `NumberFormatter` and the new API `FormatStyle` will produce the same result with respect to the locale. The latter however is shorter, simpler, and can be directly used in the SwiftUI views 🙏 
Here is the same result as yours if I use the built-in Decimal.FormatStyle.Currency:

<img width="1772" alt="Screenshot 2022-12-19 at 15 55 52" src="https://user-images.githubusercontent.com/17972674/208386645-dd0a8598-9c15-45cc-94fd-5b5551c1eb97.png">

Your result (and mine) use the local machine's current locale, which is `en_US` I believe. If I set it to another one, let's say `vi_VN`, both will change with the currency **"US$"** displayed _**after the amount**_ (with a space in between):
<img width="1593" alt="Screenshot 2022-12-19 at 15 59 21" src="https://user-images.githubusercontent.com/17972674/208387823-6f8fb7fc-9880-47b8-83d8-ce2d3aea28e1.png">
<img width="1772" alt="Screenshot 2022-12-19 at 16 02 14" src="https://user-images.githubusercontent.com/17972674/208387972-6514e30a-f927-48d4-99fb-dbedc542267b.png">

To make both displays correct, here are the modifications:
<img width="1593" alt="Screenshot 2022-12-19 at 16 13 49" src="https://user-images.githubusercontent.com/17972674/208390133-b29dea43-0c59-4222-859a-4b5a971ee642.png">
<img width="1772" alt="Screenshot 2022-12-19 at 16 13 51" src="https://user-images.githubusercontent.com/17972674/208390179-7153350b-d5eb-44f8-9195-4316a5b6faff.png">

That's why I have to make the custom `DollarCurrencyFormatStyle` which is actually just a wrapper for the built-in `Decimal.FormatStyle.number` with the `$` prefix 🙏

Reply:
@Thieurom oh I had a look through the header comments but I was unsure your mentioned `built-in Decimal.FormatStyle.Currency` relates to the `NumberFormatter` or not. 

Now everything makes sense now, thank you so much for your insights 🙏 Then I guess we can keep it as-is and just remove the redundant space between the `$` and the amount. 🚀